pandas/core/base.py

```diff
@@ -15,6 +16,24 @@
                             duplicated='IndexOpsMixin')
 
 
+def _validate_sorter(obj, sorter):
+    """
+    Checks whether or not the implementation of 'searchsorted'
+    for a given class instance 'obj' should accept a non-None
+    'sorter' argument, if one is provided. Raises a
+    NotImplementedError if that is not the case.
+    """
```

Comment:
this should be part of the ``searchsorted`` method, then just call ``super(...)`` in the subs. Again this is a difference from numpy, where we tend to have as much possible code in the same place, and avoid a litterring of functions. You can always just raise on ``sorter is not None``. doesn't matter what it is. We don't test this. If its an issue in the future it can then be implementd.